base/shell/explorer/startup.cpp

```diff
@@ -332,3 +385,85 @@ ProcessStartupItems(VOID)
 
     return res ? 0 : 101;
 }
+
+static BOOL IsSelfExplorer(VOID)
+{
+    WCHAR szPath[MAX_PATH];
+
+    GetModuleFileName(NULL, szPath, _countof(szPath));
+    PathRemoveArgsW(szPath);
+    PathRemoveBlanksW(szPath);
+
+    return (StrCmpIW(PathFindFileNameW(szPath), L"explorer.exe") == 0);
+}
+
+BOOL DoFinishStartupItems(VOID)
+{
+    if (s_hStartupMutex)
+    {
+        ReleaseMutex(s_hStartupMutex);
+        CloseHandle(s_hStartupMutex);
+        s_hStartupMutex = NULL;
+    }
+    return TRUE;
+}
+
+BOOL DoStartStartupItems(ITrayWindow *Tray)
+{
+    DWORD dwWait;
+
+    if (!IsSelfExplorer())
+        return FALSE;
+
+    if (!s_hStartupMutex)
+    {
+        // Accidentally, there is possibility that the system starts multple Explorers without shell.
```

Comment:
"multiple"

Reply:
What do you mean by "multiple explorers without shell" ? You mean, that it opens folder windows, but not the full-fledged shell (with taskbar and co.)? (might happen if the default shell is *not* explorer.)

Reply:
s/without shell/before startup of shell/

Reply:
OK.